### PR TITLE
Vulcan show multiple UI inputs + checkbox multi-select UI component

### PR DESCRIPTION
**Pull request by coleshaw**
This PR fixes some minor bugs, will render all available UI input steps at once (instead of just the next one), and also adds a `checkboxes` input component for @dtm2451 . The checkboxes input will take an array of options and basically render each as a checkbox, returning the ones left checked to the next step.

Note that the checkboxes component is currently unstyled, since I'm going to work on re-working the layout as my next step, and will just end up re-styling everything, I think.

### CODE REVIEW

**Comment by dtm2451** [comment]
Hmm trying to check but I don't see the multiselect-strings rendered & so I end up with a workflow just doesn't run.  Inspector:console gives this error:
```
Source map error: Error: request failed with status 404
Resource URL: https://vulcan.development.local/js/vulcan.bundle.js
Source Map URL: factory.js.map
```

**Comment by coleshaw** [comment]
> Hmm trying to check but I don't see the multiselect-strings rendered & so I end up with a workflow just doesn't run. Inspector:console gives this error:
> 
> ```
> Source map error: Error: request failed with status 404
> Resource URL: https://vulcan.development.local/js/vulcan.bundle.js
> Source Map URL: factory.js.map
> ```

Is there another error? That's fairly generic I think, that just means we aren't including a sourcemap for the webpacked JavaScript bundle and might show up in my working console, too. Do you see anything else in the console, or in `make -C vulcan logs`?

**Comment by coleshaw** [comment]
Also, what do you see rendered? Can you clarify what the error symptoms you see are -- do the four "Select X" steps not render at all, do they render but with regular Input boxes, etc.?

**Comment by coleshaw** [comment]
Oh, because the input names changed with the new grouping method, you may have to clear out local storage in your browser. Open up the dev console, go to Storage, click over to https://vulcan.development.local, and right-click / Delete All in the main pane.

I'll see if there is a way to check the cached session against the existing workflow, to help clear that issue up.

Screenshot from 2021-03-18 13-14-55

**Comment by dtm2451** [comment]
Works now!

File: archimedes/archimedes/functions/utils/json_utils.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import json
+
+
+def json_loads(json_string):
+    return json.loads(json_string)
```
Comment (dtm2451) [inline archimedes/archimedes/functions/utils/json_utils.py:5]: Does `import_json` from dataflow/path.py not work?

Comment (coleshaw) [reply archimedes/archimedes/functions/utils/json_utils.py:5]: Oh, thanks! I'll update.

Comment (dtm2451) [reply archimedes/archimedes/functions/utils/json_utils.py:5]: oops, *input, but you probably figured that out already lol

**Comment by dtm2451** [comment]
Had to unbreak my dev after breaking it, but seems to get through all the target steps now and only breaks in the expected, not synced, query step that follows.  :+1: from me!  It'll be really awesome to see this merged in with my recent PR that should make these multi-selects fill in as the should & actually work!

**Comment by coleshaw** [comment]
> Had to unbreak my dev after breaking it, but seems to get through all the target steps now and only breaks in the expected, not synced, query step that follows. +1 from me! It'll be really awesome to see this merged in with my recent PR that should make these multi-selects fill in as the should & actually work!

Hm, where does it break? I actually thought it should go through all the way and fill out with hardcoded record names? I can get all the steps to complete, and have a download link for the h5ad file...

**Comment by dtm2451** [comment]
Oops, sorry I'd missed your question!  In my mac boot now and won't be able to switch back in order to properly answer ~5pm.  But it broke down for me at the below step with, if I remember correctly, a parsing error.  Maybe the selections I'd made? idk
```
magma_query_paths:
    run: scripts/magma_query_paths.cwl
    label: 'Query paths to raw counts files'
```

**Comment by dtm2451** [comment]
image
```
Python error while executing script: /root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'magma.ucsf.edu'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
/root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'janus.ucsf.edu'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
/root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'dp.ucsf.edu'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
/root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'dp.ucsf.edu'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
Traceback (most recent call last):
  File "/archimedes-exec/tmprwck5q1n/script.py", line 19, in <module>
    h5_locations = magma.query(projectName=project_name, queryTerms=[seq_model_name,
  File "/root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/magby/Magby.py", line 94, in query
    content, _ = self._call_api(payload, magma)
  File "/root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/magby/Magby.py", line 155, in _call_api
    return magma.magmaCall(payload)
  File "/root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/magby/Magma.py", line 72, in magmaCall
    content = self.getResponseContent(response)
  File "/root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/magby/Magma.py", line 50, in getResponseContent
    return switch[self._fmt](response)
  File "/root/.cache/pypoetry/virtualenvs/archimedes-9TtSrW0h-py3.8/lib/python3.8/site-packages/magby/Magma.py", line 54, in _parseJSON
    return json.loads(response.text, strict=False)
  File "/usr/local/lib/python3.8/json/__init__.py", line 370, in loads
    return cls(**kw).decode(s)
  File "/usr/local/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

I think the first few lines just indicate something I haven't set up in my own linux system, but that I thought could be ignored.  Maybe it can't be though...
The real error is a broken magby magma call, but I think I have my config set up properly & you've said it works for you!

My config:
image

**Comment by dtm2451** [comment]
Yea... I get the same-ish on master now, so certainly not your branch!

**Comment by coleshaw** [comment]
I think you’ve misspelled “:archimedes_token”. Looks like it’s missing an “e”. 

Don’t worry about the warnings. They are an artifact of how I set it up for local dev and will get removed at some point.

**Comment by coleshaw** [comment]
Apologies if I put that into the template wrong! I will check and verify the spelling in the template.

**Comment by coleshaw** [comment]
@dtm2451 looks like I had a typo in the template, sorry! Fixed that, but you'll need to fix your config.yml file locally.

**Pull request merged**